### ISSUE TITLE
run `hack/update-codegen.sh` failed in MacOS

### ISSUE THREAD

**Issue by zhaojizhuang** [knative/hack#50]
It failed when run `hack/update-codegen.sh` in Mac OS

```shell
sh hack/update-codegen.sh
hack/../vendor/knative.dev/hack/library.sh: line 25: conditional binary operator expected
```

https://github.com/knative/hack/blob/master/library.sh#L25

**Comment by dprotaso** [comment]
We expect bash 4 even on Mac OS X

**Comment by zhaojizhuang** [comment]
> We expect bash 4 even on Mac OS X

yeah, Problem solved After update bash to 5.1.4,

```shell
$ bash --version
GNU bash，版本 5.1.4(1)-release (x86_64-apple-darwin19.6.0)
Copyright (C) 2020 Free Software Foundation, Inc.
```

be careful :  ` Read-only file system` occurs when rm origin bash, using follow script can avoid System Integrity Protection

```shell
sudo mount -uw /
```
